File: src/main/java/me/skrew/data/consumer/es/Log.java
package me.skrew.data.consumer.es;

import lombok.Builder;
import org.springframework.data.annotation.Id;
import org.springframework.data.elasticsearch.annotations.Document;

import java.time.LocalDateTime;

@Document(indexName = "logs")
@Builder
public class Log {

    @Id
    private Long id;
    private String summary;
    private LocalDateTime createdAt;

}

File: src/main/kotlin/me/skrew/data/consumer/es/Log.kt
package me.skrew.data.consumer.es

import org.springframework.data.annotation.Id
import org.springframework.data.elasticsearch.annotations.Document
import java.time.LocalDateTime

@Document(indexName = "logs")
class Log(
    private val summary: String,
    private val createdAt: LocalDateTime
) {
    @Id
    private val id: String? = null
}
